content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,281 @@
+    );
+    bump.sign(someAccount);
+    return server.submitTransaction(bump);
+  }
+  // ...other error conditions...
+}).then(...);
+```
+
+</CodeExample>
+
+Note an [important stipulation](https://developers.stellar.org/docs/glossary/fee-bumps/#replace-by-fee) of fee bumping that's fulfilled above:
+
+> If you submit two distinct transactions with the same source account and sequence number, and the second transaction is a fee-bump transaction, the second transaction will be included in the transaction queue in place of the first transaction if and only if **the fee bid of the second transaction is at least 10x the fee bid of the first transaction**.
+
+
+### Timeouts
+If you receive a 504 error from Horizon after a transaction submission, you've encountered a timeout. In the Stellar universe, a timeout indicates something a little nebulous and non-traditional: it's more of a *warning* that your request hasn't been fulfilled within a reasonable amount of time *yet* more than an error. This subtlety arises because of the nature of the relationship between Horizon and Stellar Core: the network might take some time to accept a transaction---especially one with a low fee, see [Surge Pricing](#surge-pricing) later---on the order of 5-10 minutes in the worst cases, whereas Horizon needs to provide developers with a response within 30 seconds or so.
+
+This leads to a very important point: **receiving a 504 for your transaction submission does not mean the transaction did not make it to the network**.
+
+There's no mechanism to "cancel" a transaction after it has been submitted aside from by exceeding its timebounds. In fact, one of the best tools available for managing transaction failures is through [timebounds](https://developers.stellar.org/docs/glossary/transactions/#time-bounds). While optional, they're *crucial* in simplifying error handling: it's much easier to know when to check a transaction's status when you know the maximum amount of time it can be stuck.
+
+With this trick in our back pocket, resolving 504s is straightforward: until the timebound has expired, you can safely retry submission *without modifying the transaction*. Horizon makes an important guarantee (excerpt is from [this example](https://developers.stellar.org/docs/tutorials/send-and-receive-payments/#send-a-payment)) for this scenario:
+
+> If the transaction has already been successfully applied to the ledger, Horizon will simply return the saved result and not attempt to submit the transaction again. Only in cases where a transaction’s status is unknown (and thus will have a chance of being included into a ledger) will a resubmission to the network occur.
+
+In other words, if your transaction made it through after the 504, you'll get your previously-submitted transaction back. If it didn't (e.g. a `tx_too_late` result code), you're essentially back to square one: you might 504 again, fail for another reason (e.g. `tx_bad_seq`), or succeed.
```

Comment:
I didn't realize `504` meant core was still deciding -- I figured it had been rejected and that was that.

This brings me to another question. Lets say I get a 504. This means my fee is above the minimum so I didn't get a `tx_insufficient_fee`, but its not competitive with the other tx's being submitted, right? Is there a non-fee-related issue that would result in getting a `504` back?

If not, then we should state that `504`s mean the transaction's was too low relative to the other transactions being submitted, and maybe explain how that differs from `tx_insufficient_fee`. 

If there are other scenarios, is there a way to figure out what the reason was, if it wasn't related to fees?

Reply:
> Is there a non-fee-related issue that would result in getting a 504 back?

Sure, there are plenty of transient issues that could cause it, I think. From my understanding, a `504 Timeout` **might** mean that Core is still deciding; it could also be due to a problem between Horizon and Core for example, or the Core node and the rest of the Stellar network. 

It's hard (if not impossible) to determine those error details, which is why tx resub until a hard success/failure is basically the only thing you can do.